src/OpenFeature/OpenFeatureClient.cs

```diff
@@ -0,0 +1,150 @@
+
+        public FlagEvaluationDetails<T> GetObjectDetails<T>(string flagKey, T defaultValue, EvaluationContext context = null, FlagEvaluationOptions config = null)  =>
+            EvaluateFlag(flagKey, defaultValue, FlagValueType.Object, context, config);
+
+
+        private FlagEvaluationDetails<T> EvaluateFlag<T>(string flagKey, T defaultValue, FlagValueType flagValueType, EvaluationContext context = null, FlagEvaluationOptions options = null)
+        {
+            var provider = OpenFeature.GetProvider();
+         
+            // New up a evaluation context if one was not provided.
+            if (context == null) 
+            {
+                context = new EvaluationContext();
+            }
+
+            /*var allHooks = new List<IHook>()
+                .Concat(OpenFeature.GetHooks())
+                .Concat(_hooks)
+                .ToList();
+            
+            var allHooksReversed = allHooks
+                .AsEnumerable()
+                .Reverse()
+                .ToList();
+            
+            var hookContext = new HookContext<bool>(
+                flagKey,
+                defaultValue,
+                FlagValueType.Boolean,
+                _metadata,
+                OpenFeature.GetProviderMetadata(),
+                context
+            );*/
+
+            FlagEvaluationDetails<T> response;
+            try
+            {
+                //TriggerHook(HookLifeCycle.Before, allHooks, hookContext, config);
+
+                // TODO Extract this out into another method so this method isn't too complex
+                switch (flagValueType)
+                {
+                    case FlagValueType.Boolean:
+                        response = provider.ResolveBooleanValue(flagKey, (bool)Convert.ChangeType(defaultValue, typeof(bool)),
+                            context, options).ToFlagEvaluationDetails() as FlagEvaluationDetails<T>;
+                        break;
+                    case FlagValueType.Number:
+                        response = provider.ResolveNumberValue(flagKey, (int)Convert.ChangeType(defaultValue, typeof(int)),
+                            context, options).ToFlagEvaluationDetails() as FlagEvaluationDetails<T>;
+                        break;
+                    case FlagValueType.String:
+                        response = provider.ResolveStringValue(flagKey, (string)Convert.ChangeType(defaultValue, typeof(string)),
+                            context, options).ToFlagEvaluationDetails() as FlagEvaluationDetails<T>;
+                        break;
+                    case FlagValueType.Object:
+                        response = provider.ResolveStructureValue(flagKey, defaultValue, context, options).ToFlagEvaluationDetails();
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException(nameof(flagValueType), flagValueType, "Unknown flag value type");
+                }

```

Comment:
You may be able to get rid of this switch case by using [delegates](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/delegates/). In your calls to this `EvaluateFlag` you can pass a ref to the `provider.ResolveXxxx` method you want to call, since the caller knows which one needs to be called. 

I did something similar in TS: https://github.com/open-feature/node-sdk/blob/98f93067d7779bf391d314af6a3c2b153523ba13/src/client.ts#L146

Up to you if you want to do this!

Reply:
Yea completely blanked about using a delegate. I've updated the implementation based on your recommendation